<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="KiseokLee" />

<meta name="date" content="2022-04-05" />

<title>220406_class_activity_import_data</title>

<script src="site_libs/header-attrs-2.9/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">QME</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/KiseokLee/QME">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">220406_class_activity_import_data</h1>
<h4 class="author">KiseokLee</h4>
<h4 class="date">2022-04-05</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-04-05
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>QME/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220403code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20220403)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220403code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220403)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomKiseokLeeQMEtree03c4d4793d77d897c2bc0fc4fbb5e804b5ef0193targetblank03c4d47a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/KiseokLee/QME/tree/03c4d4793d77d897c2bc0fc4fbb5e804b5ef0193" target="_blank">03c4d47</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomKiseokLeeQMEtree03c4d4793d77d897c2bc0fc4fbb5e804b5ef0193targetblank03c4d47a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/KiseokLee/QME/tree/03c4d4793d77d897c2bc0fc4fbb5e804b5ef0193" target="_blank">03c4d47</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  Sample.label_metadata.xlsx
    Untracked:  data/OM.CompanionTables.xlsx
    Untracked:  data/Sample.label_metadata.xlsx
    Untracked:  data/miTAG.taxonomic.profiles.release.tsv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/220406_class_activity_import_data.Rmd</code>) and HTML (<code>docs/220406_class_activity_import_data.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KiseokLee/QME/blob/03c4d4793d77d897c2bc0fc4fbb5e804b5ef0193/analysis/220406_class_activity_import_data.Rmd" target="_blank">03c4d47</a>
</td>
<td>
KiseokLee
</td>
<td>
2022-04-05
</td>
<td>
wflow_publish(“analysis/220406_class_activity_import_data.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KiseokLee/QME/1e79acdc13e693047ff1683bc8c223b20989eaff/docs/220406_class_activity_import_data.html" target="_blank">1e79acd</a>
</td>
<td>
KiseokLee
</td>
<td>
2022-04-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KiseokLee/QME/blob/2ae634860d0ec5dcb0aede4f9da68354e4155ac7/analysis/220406_class_activity_import_data.Rmd" target="_blank">2ae6348</a>
</td>
<td>
KiseokLee
</td>
<td>
2022-04-05
</td>
<td>
wflow_publish(“analysis/220406_class_activity_import_data.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KiseokLee/QME/7125eaed5b44e07f6c38f376c4b84ccc0336ae71/docs/220406_class_activity_import_data.html" target="_blank">7125eae</a>
</td>
<td>
KiseokLee
</td>
<td>
2022-04-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KiseokLee/QME/blob/a23607518288e6403048b8e43491fffb707461d4/analysis/220406_class_activity_import_data.Rmd" target="_blank">a236075</a>
</td>
<td>
KiseokLee
</td>
<td>
2022-04-05
</td>
<td>
wflow_publish("analysis/*.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="class-activity" class="section level2">
<h2>220406 Class activity</h2>
<p>TA: <strong>Kiseok Lee</strong><br />
Date: 4/6/22<br />
Class: ECEV 36500 Quantitative microbial ecology (<strong>Seppe Kuehn</strong>)</p>
<pre class="r"><code># libraries
library(dplyr)
library(ggplot2)
library(vegan)
library(ape)
library(tidyverse)
library(tidyr)
library(magrittr)
library(readxl)
library(openxlsx)
library(reshape2)
library(gtools)
library(stringr)


mytheme_2d &lt;- theme_bw() + 
  theme(text = element_text(family=&quot;serif&quot;)) +
  theme(plot.title = element_text(size = 19,hjust = 0.5, family=&quot;serif&quot;)) + 
  theme(axis.title.x = element_text(size = 17,hjust = 0.5, family=&quot;serif&quot;)) + 
  theme(axis.title.y = element_text(size = 17,hjust = 0.5, family=&quot;serif&quot;)) + 
  theme(axis.text.x = element_text(hjust = 0.5, vjust=0.3,size=13, family=&quot;serif&quot;))+
  theme(axis.text.y = element_text(size=13, family=&quot;serif&quot;))+
  # theme(panel.grid.major = element_blank()) +
  # theme(panel.grid.minor = element_blank(),panel.background=element_blank(),plot.background=element_blank()) +
  theme(axis.ticks = element_line(size = 1.1))</code></pre>
</div>
<div id="lets-import-data" class="section level2">
<h2>1. Let’s import data</h2>
<pre class="r"><code>## change accordingly to your directory

# this is metadata (env variable, diversity, biomass, etc information is here)
df_meta &lt;- openxlsx::read.xlsx(&quot;data/Sample.label_metadata.xlsx&quot;)
dim(df_meta) # 243 samples</code></pre>
<pre><code>[1] 243  38</code></pre>
<pre class="r"><code>head(df_meta)</code></pre>
<pre><code>           Sample.label PANGAEA.Sample.ID Mean_Date.[YY/MM/DD.hh:mm]* Mean_Lat*
1 TARA_004_DCM_0.22-1.6   TARA_X000000368                    40071.70  36.57300
2 TARA_004_SRF_0.22-1.6   TARA_Y200000002                    40071.43  36.55333
3 TARA_007_DCM_0.22-1.6   TARA_A200000159                    40079.70  37.03868
4 TARA_007_SRF_0.22-1.6   TARA_A200000113                    40079.63  37.02107
5 TARA_009_DCM_0.22-1.6   TARA_X000001036                    40084.57  39.07340
6 TARA_009_SRF_0.22-1.6   TARA_X000000950                    40084.57  39.07340
  Mean_Long* Mean_Depth.[m]* Mean_Temperature.[deg.C]* Mean_Salinity.[PSU]*
1  -6.538667       38.689029                  16.24633             36.55644
2  -6.566833       10.000000                  20.52510             36.63240
3   1.949833       41.740393                  17.36746                   NA
4   1.949467        7.500000                  23.82575             37.49110
5   5.859833       55.000000                  16.15193             37.82320
6   5.859833        5.784381                  23.85260             37.97510
  Mean_Oxygen.[umol/kg]* Mean_Nitrates[umol/L]* NO2.[umol/L]** PO4.[umol/L]**
1                     NA                     NA             NA             NA
2                     NA                     NA             NA             NA
3                     NA                     NA           0.00           0.01
4                     NA                     NA           0.00           0.06
5                     NA                     NA           0.02           0.02
6                     NA                     NA           0.01           0.02
  NO2NO3.[umol/L]** SI.[umol/L]** AMODIS:PAR8d,Einsteins/m-2/d-1 Okubo-Weiss
1                NA            NA                         37.643    0.007355
2                NA            NA                         39.916    0.007355
3              0.03          0.50                         29.931   -0.794335
4              0.03          0.52                         29.931   -0.794335
5              0.08          0.60                         26.661   -0.773622
6              0.03          0.75                         26.661   -0.773622
  Lyapunov_exp. grad_SST_adv retention Mean.Depth.MLD.Sigma.[m]*
1      0.023383           NA  0.000000                       3.5
2      0.023383           NA  0.000000                       3.5
3      0.000000           NA  4.916667                      18.0
4      0.000000           NA  4.916667                      18.0
5      0.082940     0.690694 13.611111                      21.0
6      0.082940     0.690694 13.611111                      21.0
  Mean.Depth.Max.Fluo.[m]* Mean.Depth.Max.N2.[m]* Mean.Depth.Max.O2.[m]*
1                 38.00000                   17.5                     NA
2                 38.00000                   17.5                     NA
3                 44.33333                   40.5                     NA
4                 44.33333                   40.5                     NA
5                 52.50000                   34.5                     NA
6                 52.50000                   34.5                     NA
  Mean.Depth.Min.O2.[m]* Mean.Depth.Nitracline.[m]* miTAG.SILVA.Taxo.Richness
1                     NA                         NA                   3054.63
2                     NA                         NA                   2717.74
3                     NA                         NA                   3134.91
4                     NA                         NA                   2591.99
5                     NA                         NA                   3228.86
6                     NA                         NA                   2740.60
  miTAG.SILVA.Phylo.Diversity miTAG.SILVA.Chao miTAG.SILVA.ace
1                    451.0603         4173.587        4030.463
2                    406.1298         3762.955        3680.078
3                    483.2581         4786.076        4585.078
4                    400.9745         3872.353        3729.129
5                    473.3118         4462.686        4296.565
6                    413.5093         3982.790        3865.446
  miTAG.SILVA.Shannon OG.Shannon OG.Richness OG.Evenness
1            6.849891   7.160451     14607.3   0.7467149
2            6.635941   7.145722     14590.9   0.7452660
3            6.801171   7.279688     16416.5   0.7500163
4            6.525141   7.201536     15495.1   0.7464064
5            6.907613   7.154407     15318.4   0.7424042
6            6.624968   7.222911     15315.1   0.7495296
  FC.-.heterotrophs.[cells/mL] FC.-.autotrophs.[cells/mL]
1                    1226212.2                   56383.47
2                     839530.6                   54593.19
3                    2580789.0                  156124.58
4                    1004684.1                   50585.64
5                     781038.7                   19831.96
6                     901463.2                   39329.53
  FC.-.bacteria.[cells/mL] FC.-.picoeukaryotes.[cells/mL]
1                1282595.7                      6889.9384
2                 894123.8                      1221.8898
3                2736913.6                      1859.1433
4                1055269.8                       812.6444
5                 800870.7                      2882.2568
6                 940792.7                      1230.6593
  minimum.generation.time.[h]
1                    11.12740
2                    15.23070
3                    14.37019
4                    14.28710
5                    10.69975
6                    17.47472</code></pre>
<pre class="r"><code># this is the OTU read abundance in each sample (OTU is the Operational Taxonomy Unit, think of it as putative species)
df_sample &lt;- read_tsv(&quot;data/miTAG.taxonomic.profiles.release.tsv&quot;)
dim(df_sample) # 139 sample</code></pre>
<pre><code>[1] 35651   146</code></pre>
<pre class="r"><code>head(df_sample)</code></pre>
<pre><code># A tibble: 6 x 146
  Domain  Phylum   Class    Order   Family  Genus    OTU.rep   `TARA_018_DCM_0.~
  &lt;chr&gt;   &lt;chr&gt;    &lt;chr&gt;    &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;    &lt;chr&gt;                 &lt;dbl&gt;
1 undef   undef    undef    undef   undef   undef    unclassi~              5101
2 Bacter~ Proteob~ Alphapr~ Rhodos~ Rhodos~ AEGEAN-~ AACY0241~              1021
3 Bacter~ Cyanoba~ Cyanoba~ Subsec~ FamilyI Prochlo~ KC003383~              2697
4 Bacter~ Proteob~ Alphapr~ Rhodos~ Rhodos~ AEGEAN-~ EU394547~               605
5 Bacter~ Cyanoba~ Cyanoba~ Subsec~ FamilyI Prochlo~ X52169.1~              3786
6 Bacter~ Cyanoba~ Cyanoba~ Subsec~ FamilyI Prochlo~ EU802966~               530
# ... with 138 more variables: TARA_018_SRF_0.22-1.6 &lt;dbl&gt;,
#   TARA_023_DCM_0.22-1.6 &lt;dbl&gt;, TARA_023_SRF_0.22-1.6 &lt;dbl&gt;,
#   TARA_025_DCM_0.22-1.6 &lt;dbl&gt;, TARA_025_SRF_0.22-1.6 &lt;dbl&gt;,
#   TARA_030_DCM_0.22-1.6 &lt;dbl&gt;, TARA_030_SRF_0.22-1.6 &lt;dbl&gt;,
#   TARA_031_SRF_0.22-1.6 &lt;dbl&gt;, TARA_032_DCM_0.22-1.6 &lt;dbl&gt;,
#   TARA_032_SRF_0.22-1.6 &lt;dbl&gt;, TARA_033_SRF_0.22-1.6 &lt;dbl&gt;,
#   TARA_034_DCM_0.22-1.6 &lt;dbl&gt;, TARA_034_SRF_0.22-1.6 &lt;dbl&gt;,
#   TARA_036_DCM_0.22-1.6 &lt;dbl&gt;, TARA_036_SRF_0.22-1.6 &lt;dbl&gt;,
#   TARA_037_MES_0.22-1.6 &lt;dbl&gt;, TARA_038_DCM_0.22-1.6 &lt;dbl&gt;,
#   TARA_038_MES_0.22-1.6 &lt;dbl&gt;, TARA_038_SRF_0.22-1.6 &lt;dbl&gt;,
#   TARA_039_DCM_0.22-1.6 &lt;dbl&gt;, TARA_039_MES_0.22-1.6 &lt;dbl&gt;,
#   TARA_041_DCM_0.22-1.6 &lt;dbl&gt;, TARA_041_SRF_0.22-1.6 &lt;dbl&gt;,
#   TARA_042_DCM_0.22-1.6 &lt;dbl&gt;, TARA_042_SRF_0.22-1.6 &lt;dbl&gt;,
#   TARA_045_SRF_0.22-1.6 &lt;dbl&gt;, TARA_048_SRF_0.22-1.6 &lt;dbl&gt;,
#   TARA_004_DCM_0.22-1.6 &lt;dbl&gt;, TARA_004_SRF_0.22-1.6 &lt;dbl&gt;,
#   TARA_052_DCM_0.22-1.6 &lt;dbl&gt;, TARA_052_SRF_0.22-1.6 &lt;dbl&gt;,
#   TARA_007_DCM_0.22-1.6 &lt;dbl&gt;, TARA_007_SRF_0.22-1.6 &lt;dbl&gt;,
#   TARA_009_DCM_0.22-1.6 &lt;dbl&gt;, TARA_009_SRF_0.22-1.6 &lt;dbl&gt;,
#   TARA_100_DCM_0.22-3 &lt;dbl&gt;, TARA_100_MES_0.22-3 &lt;dbl&gt;,
#   TARA_100_SRF_0.22-3 &lt;dbl&gt;, TARA_102_DCM_0.22-3 &lt;dbl&gt;,
#   TARA_102_MES_0.22-3 &lt;dbl&gt;, TARA_102_SRF_0.22-3 &lt;dbl&gt;,
#   TARA_109_DCM_0.22-3 &lt;dbl&gt;, TARA_109_MES_0.22-3 &lt;dbl&gt;,
#   TARA_109_SRF_0.22-3 &lt;dbl&gt;, TARA_110_DCM_0.22-3 &lt;dbl&gt;,
#   TARA_110_MES_0.22-3 &lt;dbl&gt;, TARA_110_SRF_0.22-3 &lt;dbl&gt;,
#   TARA_111_DCM_0.22-3 &lt;dbl&gt;, TARA_111_MES_0.22-3 &lt;dbl&gt;,
#   TARA_111_SRF_0.22-3 &lt;dbl&gt;, TARA_112_DCM_0.22-3 &lt;dbl&gt;,
#   TARA_112_MES_0.22-3 &lt;dbl&gt;, TARA_112_SRF_0.22-3 &lt;dbl&gt;,
#   TARA_122_DCM_0.22-3 &lt;dbl&gt;, TARA_122_MES_0.22-3 &lt;dbl&gt;,
#   TARA_122_SRF_0.22-3 &lt;dbl&gt;, TARA_123_MIX_0.22-3 &lt;dbl&gt;,
#   TARA_123_SRF_0.22-3 &lt;dbl&gt;, TARA_124_MIX_0.22-3 &lt;dbl&gt;,
#   TARA_124_SRF_0.22-3 &lt;dbl&gt;, TARA_125_MIX_0.22-3 &lt;dbl&gt;,
#   TARA_125_SRF_0.22-3 &lt;dbl&gt;, TARA_128_DCM_0.22-3 &lt;dbl&gt;,
#   TARA_128_SRF_0.22-3 &lt;dbl&gt;, TARA_132_DCM_0.22-3 &lt;dbl&gt;,
#   TARA_132_MES_0.22-3 &lt;dbl&gt;, TARA_132_SRF_0.22-3 &lt;dbl&gt;,
#   TARA_133_DCM_0.22-3 &lt;dbl&gt;, TARA_133_MES_0.22-3 &lt;dbl&gt;,
#   TARA_133_SRF_0.22-3 &lt;dbl&gt;, TARA_137_DCM_0.22-3 &lt;dbl&gt;,
#   TARA_137_MES_0.22-3 &lt;dbl&gt;, TARA_137_SRF_0.22-3 &lt;dbl&gt;,
#   TARA_138_DCM_0.22-3 &lt;dbl&gt;, TARA_138_MES_0.22-3 &lt;dbl&gt;,
#   TARA_138_SRF_0.22-3 &lt;dbl&gt;, TARA_140_SRF_0.22-3 &lt;dbl&gt;,
#   TARA_141_SRF_0.22-3 &lt;dbl&gt;, TARA_142_DCM_0.22-3 &lt;dbl&gt;,
#   TARA_142_MES_0.22-3 &lt;dbl&gt;, TARA_142_SRF_0.22-3 &lt;dbl&gt;,
#   TARA_145_MES_0.22-3 &lt;dbl&gt;, TARA_145_SRF_0.22-3 &lt;dbl&gt;,
#   TARA_146_MES_0.22-3 &lt;dbl&gt;, TARA_146_SRF_0.22-3 &lt;dbl&gt;,
#   TARA_148b_MES_0.22-3 &lt;dbl&gt;, TARA_148_SRF_0.22-3 &lt;dbl&gt;,
#   TARA_149_MES_0.22-3 &lt;dbl&gt;, TARA_149_SRF_0.22-3 &lt;dbl&gt;,
#   TARA_150_DCM_0.22-3 &lt;dbl&gt;, TARA_150_SRF_0.22-3 &lt;dbl&gt;,
#   TARA_151_DCM_0.22-3 &lt;dbl&gt;, TARA_151_SRF_0.22-3 &lt;dbl&gt;,
#   TARA_152_MIX_0.22-3 &lt;dbl&gt;, TARA_152_MES_0.22-3 &lt;dbl&gt;,
#   TARA_152_SRF_0.22-3 &lt;dbl&gt;, TARA_056_MES_0.22-3 &lt;dbl&gt;,
#   TARA_056_SRF_0.22-3 &lt;dbl&gt;, TARA_057_SRF_0.22-3 &lt;dbl&gt;,
#   TARA_058_DCM_0.22-3 &lt;dbl&gt;, ...</code></pre>
<pre class="r"><code>colnames(df_sample)</code></pre>
<pre><code>  [1] &quot;Domain&quot;                &quot;Phylum&quot;                &quot;Class&quot;                
  [4] &quot;Order&quot;                 &quot;Family&quot;                &quot;Genus&quot;                
  [7] &quot;OTU.rep&quot;               &quot;TARA_018_DCM_0.22-1.6&quot; &quot;TARA_018_SRF_0.22-1.6&quot;
 [10] &quot;TARA_023_DCM_0.22-1.6&quot; &quot;TARA_023_SRF_0.22-1.6&quot; &quot;TARA_025_DCM_0.22-1.6&quot;
 [13] &quot;TARA_025_SRF_0.22-1.6&quot; &quot;TARA_030_DCM_0.22-1.6&quot; &quot;TARA_030_SRF_0.22-1.6&quot;
 [16] &quot;TARA_031_SRF_0.22-1.6&quot; &quot;TARA_032_DCM_0.22-1.6&quot; &quot;TARA_032_SRF_0.22-1.6&quot;
 [19] &quot;TARA_033_SRF_0.22-1.6&quot; &quot;TARA_034_DCM_0.22-1.6&quot; &quot;TARA_034_SRF_0.22-1.6&quot;
 [22] &quot;TARA_036_DCM_0.22-1.6&quot; &quot;TARA_036_SRF_0.22-1.6&quot; &quot;TARA_037_MES_0.22-1.6&quot;
 [25] &quot;TARA_038_DCM_0.22-1.6&quot; &quot;TARA_038_MES_0.22-1.6&quot; &quot;TARA_038_SRF_0.22-1.6&quot;
 [28] &quot;TARA_039_DCM_0.22-1.6&quot; &quot;TARA_039_MES_0.22-1.6&quot; &quot;TARA_041_DCM_0.22-1.6&quot;
 [31] &quot;TARA_041_SRF_0.22-1.6&quot; &quot;TARA_042_DCM_0.22-1.6&quot; &quot;TARA_042_SRF_0.22-1.6&quot;
 [34] &quot;TARA_045_SRF_0.22-1.6&quot; &quot;TARA_048_SRF_0.22-1.6&quot; &quot;TARA_004_DCM_0.22-1.6&quot;
 [37] &quot;TARA_004_SRF_0.22-1.6&quot; &quot;TARA_052_DCM_0.22-1.6&quot; &quot;TARA_052_SRF_0.22-1.6&quot;
 [40] &quot;TARA_007_DCM_0.22-1.6&quot; &quot;TARA_007_SRF_0.22-1.6&quot; &quot;TARA_009_DCM_0.22-1.6&quot;
 [43] &quot;TARA_009_SRF_0.22-1.6&quot; &quot;TARA_100_DCM_0.22-3&quot;   &quot;TARA_100_MES_0.22-3&quot;  
 [46] &quot;TARA_100_SRF_0.22-3&quot;   &quot;TARA_102_DCM_0.22-3&quot;   &quot;TARA_102_MES_0.22-3&quot;  
 [49] &quot;TARA_102_SRF_0.22-3&quot;   &quot;TARA_109_DCM_0.22-3&quot;   &quot;TARA_109_MES_0.22-3&quot;  
 [52] &quot;TARA_109_SRF_0.22-3&quot;   &quot;TARA_110_DCM_0.22-3&quot;   &quot;TARA_110_MES_0.22-3&quot;  
 [55] &quot;TARA_110_SRF_0.22-3&quot;   &quot;TARA_111_DCM_0.22-3&quot;   &quot;TARA_111_MES_0.22-3&quot;  
 [58] &quot;TARA_111_SRF_0.22-3&quot;   &quot;TARA_112_DCM_0.22-3&quot;   &quot;TARA_112_MES_0.22-3&quot;  
 [61] &quot;TARA_112_SRF_0.22-3&quot;   &quot;TARA_122_DCM_0.22-3&quot;   &quot;TARA_122_MES_0.22-3&quot;  
 [64] &quot;TARA_122_SRF_0.22-3&quot;   &quot;TARA_123_MIX_0.22-3&quot;   &quot;TARA_123_SRF_0.22-3&quot;  
 [67] &quot;TARA_124_MIX_0.22-3&quot;   &quot;TARA_124_SRF_0.22-3&quot;   &quot;TARA_125_MIX_0.22-3&quot;  
 [70] &quot;TARA_125_SRF_0.22-3&quot;   &quot;TARA_128_DCM_0.22-3&quot;   &quot;TARA_128_SRF_0.22-3&quot;  
 [73] &quot;TARA_132_DCM_0.22-3&quot;   &quot;TARA_132_MES_0.22-3&quot;   &quot;TARA_132_SRF_0.22-3&quot;  
 [76] &quot;TARA_133_DCM_0.22-3&quot;   &quot;TARA_133_MES_0.22-3&quot;   &quot;TARA_133_SRF_0.22-3&quot;  
 [79] &quot;TARA_137_DCM_0.22-3&quot;   &quot;TARA_137_MES_0.22-3&quot;   &quot;TARA_137_SRF_0.22-3&quot;  
 [82] &quot;TARA_138_DCM_0.22-3&quot;   &quot;TARA_138_MES_0.22-3&quot;   &quot;TARA_138_SRF_0.22-3&quot;  
 [85] &quot;TARA_140_SRF_0.22-3&quot;   &quot;TARA_141_SRF_0.22-3&quot;   &quot;TARA_142_DCM_0.22-3&quot;  
 [88] &quot;TARA_142_MES_0.22-3&quot;   &quot;TARA_142_SRF_0.22-3&quot;   &quot;TARA_145_MES_0.22-3&quot;  
 [91] &quot;TARA_145_SRF_0.22-3&quot;   &quot;TARA_146_MES_0.22-3&quot;   &quot;TARA_146_SRF_0.22-3&quot;  
 [94] &quot;TARA_148b_MES_0.22-3&quot;  &quot;TARA_148_SRF_0.22-3&quot;   &quot;TARA_149_MES_0.22-3&quot;  
 [97] &quot;TARA_149_SRF_0.22-3&quot;   &quot;TARA_150_DCM_0.22-3&quot;   &quot;TARA_150_SRF_0.22-3&quot;  
[100] &quot;TARA_151_DCM_0.22-3&quot;   &quot;TARA_151_SRF_0.22-3&quot;   &quot;TARA_152_MIX_0.22-3&quot;  
[103] &quot;TARA_152_MES_0.22-3&quot;   &quot;TARA_152_SRF_0.22-3&quot;   &quot;TARA_056_MES_0.22-3&quot;  
[106] &quot;TARA_056_SRF_0.22-3&quot;   &quot;TARA_057_SRF_0.22-3&quot;   &quot;TARA_058_DCM_0.22-3&quot;  
[109] &quot;TARA_062_SRF_0.22-3&quot;   &quot;TARA_064_DCM_0.22-3&quot;   &quot;TARA_064_MES_0.22-3&quot;  
[112] &quot;TARA_064_SRF_0.22-3&quot;   &quot;TARA_065_DCM_0.22-3&quot;   &quot;TARA_065_MES_0.22-3&quot;  
[115] &quot;TARA_065_SRF_0.22-3&quot;   &quot;TARA_066_DCM_0.22-3&quot;   &quot;TARA_066_SRF_0.22-3&quot;  
[118] &quot;TARA_067_SRF_0.22-3&quot;   &quot;TARA_068_DCM_0.22-3&quot;   &quot;TARA_068_MES_0.22-3&quot;  
[121] &quot;TARA_068_SRF_0.22-3&quot;   &quot;TARA_070_MES_0.22-3&quot;   &quot;TARA_070_SRF_0.22-3&quot;  
[124] &quot;TARA_072_DCM_0.22-3&quot;   &quot;TARA_072_MES_0.22-3&quot;   &quot;TARA_072_SRF_0.22-3&quot;  
[127] &quot;TARA_076_DCM_0.22-3&quot;   &quot;TARA_076_MES_0.22-3&quot;   &quot;TARA_076_SRF_0.22-3&quot;  
[130] &quot;TARA_078_DCM_0.22-3&quot;   &quot;TARA_078_MES_0.22-3&quot;   &quot;TARA_078_SRF_0.22-3&quot;  
[133] &quot;TARA_082_DCM_0.22-3&quot;   &quot;TARA_082_SRF_0.22-3&quot;   &quot;TARA_084_SRF_0.22-3&quot;  
[136] &quot;TARA_085_DCM_0.22-3&quot;   &quot;TARA_085_MES_0.22-3&quot;   &quot;TARA_085_SRF_0.22-3&quot;  
[139] &quot;TARA_093_DCM_0.22-3&quot;   &quot;TARA_093_SRF_0.22-3&quot;   &quot;TARA_094_SRF_0.22-3&quot;  
[142] &quot;TARA_096_SRF_0.22-3&quot;   &quot;TARA_098_DCM_0.22-3&quot;   &quot;TARA_098_MES_0.22-3&quot;  
[145] &quot;TARA_098_SRF_0.22-3&quot;   &quot;TARA_099_SRF_0.22-3&quot;  </code></pre>
<pre class="r"><code># Agglomerate reads on taxonomic levels
vec_sample &lt;- colnames(df_sample)[grepl(&quot;TARA*&quot;,colnames(df_sample))]
length(vec_sample) # 139 samples</code></pre>
<pre><code>[1] 139</code></pre>
<pre class="r"><code>taxa_level = &quot;Class&quot;
df_taxa_level &lt;- df_sample %&gt;% select(taxa_level, vec_sample)

# add up the read abundance in each taxa in each sample
df_taxa_aggre &lt;- aggregate(df_taxa_level[,2:ncol(df_taxa_level)], by = df_taxa_level[1], function(x) sum(x, na.rm = TRUE))
dim(df_taxa_aggre)</code></pre>
<pre><code>[1] 147 140</code></pre>
<pre class="r"><code>#head(df_taxa_aggre)

# get relative abundance
df_rel_abun &lt;- df_taxa_aggre
df_rel_abun &lt;- tibble::column_to_rownames(df_rel_abun, var=taxa_level)
df_rel_abun &lt;- apply(df_rel_abun, 2, function(i) i/sum(i))
apply(df_rel_abun, 2, sum) # sum of each sample is 1</code></pre>
<pre><code>TARA_018_DCM_0.22-1.6 TARA_018_SRF_0.22-1.6 TARA_023_DCM_0.22-1.6 
                    1                     1                     1 
TARA_023_SRF_0.22-1.6 TARA_025_DCM_0.22-1.6 TARA_025_SRF_0.22-1.6 
                    1                     1                     1 
TARA_030_DCM_0.22-1.6 TARA_030_SRF_0.22-1.6 TARA_031_SRF_0.22-1.6 
                    1                     1                     1 
TARA_032_DCM_0.22-1.6 TARA_032_SRF_0.22-1.6 TARA_033_SRF_0.22-1.6 
                    1                     1                     1 
TARA_034_DCM_0.22-1.6 TARA_034_SRF_0.22-1.6 TARA_036_DCM_0.22-1.6 
                    1                     1                     1 
TARA_036_SRF_0.22-1.6 TARA_037_MES_0.22-1.6 TARA_038_DCM_0.22-1.6 
                    1                     1                     1 
TARA_038_MES_0.22-1.6 TARA_038_SRF_0.22-1.6 TARA_039_DCM_0.22-1.6 
                    1                     1                     1 
TARA_039_MES_0.22-1.6 TARA_041_DCM_0.22-1.6 TARA_041_SRF_0.22-1.6 
                    1                     1                     1 
TARA_042_DCM_0.22-1.6 TARA_042_SRF_0.22-1.6 TARA_045_SRF_0.22-1.6 
                    1                     1                     1 
TARA_048_SRF_0.22-1.6 TARA_004_DCM_0.22-1.6 TARA_004_SRF_0.22-1.6 
                    1                     1                     1 
TARA_052_DCM_0.22-1.6 TARA_052_SRF_0.22-1.6 TARA_007_DCM_0.22-1.6 
                    1                     1                     1 
TARA_007_SRF_0.22-1.6 TARA_009_DCM_0.22-1.6 TARA_009_SRF_0.22-1.6 
                    1                     1                     1 
  TARA_100_DCM_0.22-3   TARA_100_MES_0.22-3   TARA_100_SRF_0.22-3 
                    1                     1                     1 
  TARA_102_DCM_0.22-3   TARA_102_MES_0.22-3   TARA_102_SRF_0.22-3 
                    1                     1                     1 
  TARA_109_DCM_0.22-3   TARA_109_MES_0.22-3   TARA_109_SRF_0.22-3 
                    1                     1                     1 
  TARA_110_DCM_0.22-3   TARA_110_MES_0.22-3   TARA_110_SRF_0.22-3 
                    1                     1                     1 
  TARA_111_DCM_0.22-3   TARA_111_MES_0.22-3   TARA_111_SRF_0.22-3 
                    1                     1                     1 
  TARA_112_DCM_0.22-3   TARA_112_MES_0.22-3   TARA_112_SRF_0.22-3 
                    1                     1                     1 
  TARA_122_DCM_0.22-3   TARA_122_MES_0.22-3   TARA_122_SRF_0.22-3 
                    1                     1                     1 
  TARA_123_MIX_0.22-3   TARA_123_SRF_0.22-3   TARA_124_MIX_0.22-3 
                    1                     1                     1 
  TARA_124_SRF_0.22-3   TARA_125_MIX_0.22-3   TARA_125_SRF_0.22-3 
                    1                     1                     1 
  TARA_128_DCM_0.22-3   TARA_128_SRF_0.22-3   TARA_132_DCM_0.22-3 
                    1                     1                     1 
  TARA_132_MES_0.22-3   TARA_132_SRF_0.22-3   TARA_133_DCM_0.22-3 
                    1                     1                     1 
  TARA_133_MES_0.22-3   TARA_133_SRF_0.22-3   TARA_137_DCM_0.22-3 
                    1                     1                     1 
  TARA_137_MES_0.22-3   TARA_137_SRF_0.22-3   TARA_138_DCM_0.22-3 
                    1                     1                     1 
  TARA_138_MES_0.22-3   TARA_138_SRF_0.22-3   TARA_140_SRF_0.22-3 
                    1                     1                     1 
  TARA_141_SRF_0.22-3   TARA_142_DCM_0.22-3   TARA_142_MES_0.22-3 
                    1                     1                     1 
  TARA_142_SRF_0.22-3   TARA_145_MES_0.22-3   TARA_145_SRF_0.22-3 
                    1                     1                     1 
  TARA_146_MES_0.22-3   TARA_146_SRF_0.22-3  TARA_148b_MES_0.22-3 
                    1                     1                     1 
  TARA_148_SRF_0.22-3   TARA_149_MES_0.22-3   TARA_149_SRF_0.22-3 
                    1                     1                     1 
  TARA_150_DCM_0.22-3   TARA_150_SRF_0.22-3   TARA_151_DCM_0.22-3 
                    1                     1                     1 
  TARA_151_SRF_0.22-3   TARA_152_MIX_0.22-3   TARA_152_MES_0.22-3 
                    1                     1                     1 
  TARA_152_SRF_0.22-3   TARA_056_MES_0.22-3   TARA_056_SRF_0.22-3 
                    1                     1                     1 
  TARA_057_SRF_0.22-3   TARA_058_DCM_0.22-3   TARA_062_SRF_0.22-3 
                    1                     1                     1 
  TARA_064_DCM_0.22-3   TARA_064_MES_0.22-3   TARA_064_SRF_0.22-3 
                    1                     1                     1 
  TARA_065_DCM_0.22-3   TARA_065_MES_0.22-3   TARA_065_SRF_0.22-3 
                    1                     1                     1 
  TARA_066_DCM_0.22-3   TARA_066_SRF_0.22-3   TARA_067_SRF_0.22-3 
                    1                     1                     1 
  TARA_068_DCM_0.22-3   TARA_068_MES_0.22-3   TARA_068_SRF_0.22-3 
                    1                     1                     1 
  TARA_070_MES_0.22-3   TARA_070_SRF_0.22-3   TARA_072_DCM_0.22-3 
                    1                     1                     1 
  TARA_072_MES_0.22-3   TARA_072_SRF_0.22-3   TARA_076_DCM_0.22-3 
                    1                     1                     1 
  TARA_076_MES_0.22-3   TARA_076_SRF_0.22-3   TARA_078_DCM_0.22-3 
                    1                     1                     1 
  TARA_078_MES_0.22-3   TARA_078_SRF_0.22-3   TARA_082_DCM_0.22-3 
                    1                     1                     1 
  TARA_082_SRF_0.22-3   TARA_084_SRF_0.22-3   TARA_085_DCM_0.22-3 
                    1                     1                     1 
  TARA_085_MES_0.22-3   TARA_085_SRF_0.22-3   TARA_093_DCM_0.22-3 
                    1                     1                     1 
  TARA_093_SRF_0.22-3   TARA_094_SRF_0.22-3   TARA_096_SRF_0.22-3 
                    1                     1                     1 
  TARA_098_DCM_0.22-3   TARA_098_MES_0.22-3   TARA_098_SRF_0.22-3 
                    1                     1                     1 
  TARA_099_SRF_0.22-3 
                    1 </code></pre>
<pre class="r"><code># function for getting relative abundance dataframe for each taxa level
make_df_rel_abun &lt;- function(df_sample, taxa_level = &quot;Class&quot;){
  # Agglomerate reads on taxonomic levels
  vec_sample &lt;- colnames(df_sample)[grepl(&quot;TARA*&quot;,colnames(df_sample))]
  length(vec_sample) # 139 samples
  
  df_taxa_level &lt;- df_sample %&gt;% select(taxa_level, vec_sample)
  
  # add up the read abundance in each taxa in each sample
  df_taxa_aggre &lt;- aggregate(df_taxa_level[,2:ncol(df_taxa_level)], by = df_taxa_level[1], function(x) sum(x, na.rm = TRUE))
  dim(df_taxa_aggre)
  head(df_taxa_aggre)
  
  # get relative abundance
  df_rel_abun &lt;- df_taxa_aggre
  df_rel_abun &lt;- tibble::column_to_rownames(df_rel_abun, var=taxa_level)
  df_rel_abun &lt;- apply(df_rel_abun, 2, function(i) i/sum(i))
  apply(df_rel_abun, 2, sum) # sum of each sample is 1
  
  return(df_rel_abun)
}

# for all taxa level
df_phylum &lt;- make_df_rel_abun(df_sample, taxa_level = &quot;Phylum&quot;) 
df_class &lt;- make_df_rel_abun(df_sample, taxa_level = &quot;Class&quot;) 
df_order &lt;- make_df_rel_abun(df_sample, taxa_level = &quot;Order&quot;) 
df_family &lt;- make_df_rel_abun(df_sample, taxa_level = &quot;Family&quot;) 
df_genus &lt;- make_df_rel_abun(df_sample, taxa_level = &quot;Genus&quot;) 
df_otu &lt;- df_sample %&gt;% select(OTU.rep, vec_sample)</code></pre>
</div>
<div id="ordination-analysis" class="section level2">
<h2>2. Ordination analysis</h2>
</div>
<div id="normalization-sunagawa-2015" class="section level2">
<h2>2.1. Normalization (Sunagawa, 2015)</h2>
<p>For compositional data, we applied a logarithmic transformation to relative abundances using the function log10(x + x0), where x is the original relative abundance and x0 is a small constant, and x0 &lt; min(x).</p>
<pre class="r"><code># which taxa level?
df_x = df_class

# do we have any NA in the matrix?
any(is.na(df_x)) # nope</code></pre>
<pre><code>[1] FALSE</code></pre>
<pre class="r"><code># should we do normalization of relative abundance before ordination?
# Let&#39;s do the same as the Sunagawa 2015 paper.
x0 = min(df_x[df_x &gt;0]) * 0.001 
df_x_lognorm &lt;- log10(df_x + x0)</code></pre>
</div>
<div id="pca-method" class="section level2">
<h2>2.2. PCA method</h2>
<pre class="r"><code># PCA with prcomp
pc &lt;- prcomp(t(df_x_lognorm)) # transpose
eigs &lt;- pc$sdev^2
# variance explained
eigs[1] / sum(eigs) # pc1</code></pre>
<pre><code>[1] 0.3582933</code></pre>
<pre class="r"><code>eigs[2] / sum(eigs) # pc2</code></pre>
<pre><code>[1] 0.05175907</code></pre>
<pre class="r"><code># dataframe for plotting
df_out &lt;- as.data.frame(pc$x) # extract principal components
dim(df_out)</code></pre>
<pre><code>[1] 139 139</code></pre>
<pre class="r"><code>#head(df_out)

# add environmental variable
df_out &lt;- df_out %&gt;% select(PC1, PC2, PC3) # get only PC1, 2, 3
df_out &lt;- tibble::rownames_to_column(df_out, var = &quot;Sample.label&quot;)
dim(df_out)</code></pre>
<pre><code>[1] 139   4</code></pre>
<pre class="r"><code># get depth
# split sample label to get the depth SRF, DCM, MES
list_depth &lt;- str_split(df_out$Sample.label, &quot;_&quot;)
vec_depth &lt;- c()
for (i in 1:length(list_depth)){
  #print(i)
  # print(list_depth[[i]][3])
  vec_depth[i] &lt;- list_depth[[i]][3]
}
  
df_out$Depth &lt;- vec_depth
  
df_merged &lt;- df_out %&gt;% left_join(df_meta, by = c(&quot;Sample.label&quot;=&quot;Sample.label&quot;))
dim(df_merged)</code></pre>
<pre><code>[1] 139  42</code></pre>
<pre class="r"><code># colnames(df_merged)

# plot PCA
ggplot(df_merged,aes(x=PC1,y=PC2, label = Depth))+
  theme(plot.title = element_text(size = 20,hjust = 0.5, face=&quot;bold&quot;)) +
  geom_point(size = 5, alpha=0.9, aes(col= Depth))+
  scale_fill_manual(values=cols) +
  xlab(paste0(&#39;\n PC1&#39;, &quot; (explained variance: &quot;,round(eigs[1] / sum(eigs),3)*100,&quot;%)&quot;))+
  ylab(paste0(&quot;PC2&quot;,&quot; (explained variance: &quot;,round(eigs[2] / sum(eigs),3)*100,&quot;%) \n&quot;)) +
  ggtitle(&quot;Principle component analysis for all samples \n&quot;) +
  ## adjust positions
  guides(fill = guide_legend(ncol = 4,reverse = T))+
  theme(legend.position=&quot;bottom&quot;) +
  mytheme_2d</code></pre>
<p><img src="figure/220406_class_activity_import_data.Rmd/unnamed-chunk-4-1.png" width="1056" style="display: block; margin: auto;" /></p>
</div>
<div id="regress-pc1-and-pc2-on-to-environmental-variables-etc." class="section level2">
<h2>2.2.1. regress PC1 and PC2 on to environmental variables, etc.</h2>
<ul>
<li><a href="https://www.rdocumentation.org/packages/vegan/versions/2.4-2/topics/envfit" class="uri">https://www.rdocumentation.org/packages/vegan/versions/2.4-2/topics/envfit</a><br />
The function fits environmental vectors or factors onto an ordination. The projections of points onto vectors have maximum correlation with corresponding environmental variables, and the factors show the averages of factor levels. For continuous varaibles this is equal to fitting a linear trend surface (plane in 2D) for a variable (see ordisurf); this trend surface can be presented by showing its gradient (direction of steepest increase) using an arrow. The environmental variables are the dependent variables that are explained by the ordination scores, and each dependent variable is analysed separately.<br />
</li>
<li>How this is calculated: <a href="https://stackoverflow.com/questions/60953996/how-are-envfit-results-created" class="uri">https://stackoverflow.com/questions/60953996/how-are-envfit-results-created</a><br />
</li>
<li><a href="https://www.davidzeleny.net/anadat-r/doku.php/en:suppl_vars_examples" class="uri">https://www.davidzeleny.net/anadat-r/doku.php/en:suppl_vars_examples</a><br />
</li>
<li><a href="https://www.davidzeleny.net/anadat-r/doku.php/en:pca_examples" class="uri">https://www.davidzeleny.net/anadat-r/doku.php/en:pca_examples</a><br />
</li>
<li>Visualization: <a href="https://jkzorz.github.io/2020/04/04/NMDS-extras.html" class="uri">https://jkzorz.github.io/2020/04/04/NMDS-extras.html</a><br />
</li>
</ul>
<pre class="r"><code>## Let&#39;s use envfit or vectorfit function to fit environmental variable/trait to PC1 and PC2

# PCA with prcomp
df_meta1 &lt;- tibble::remove_rownames(df_meta)
df_meta1 &lt;- df_meta1 %&gt;% filter(Sample.label %in% df_out$Sample.label)
df_meta1 &lt;- tibble::column_to_rownames(df_meta1, var = &quot;Sample.label&quot;)
df_meta1 &lt;- df_meta1[,c(-1,-2)] # remove PANGAEA id and mean date
#head(df_meta1)
df_meta_norm &lt;- scale(df_meta1) # z-score transformation
df_meta_norm &lt;- as.data.frame(df_meta_norm)
apply(df_meta_norm, 2, mean, na.rm=T) # column wise mean is 0</code></pre>
<pre><code>                     Mean_Lat*                     Mean_Long* 
                 -3.261499e-17                  -3.165597e-17 
               Mean_Depth.[m]*      Mean_Temperature.[deg.C]* 
                  5.788938e-17                  -2.333468e-16 
          Mean_Salinity.[PSU]*         Mean_Oxygen.[umol/kg]* 
                  1.299127e-15                   2.146076e-16 
        Mean_Nitrates[umol/L]*                 NO2.[umol/L]** 
                 -1.168108e-17                  -2.674586e-17 
                PO4.[umol/L]**              NO2NO3.[umol/L]** 
                 -3.331894e-17                  -3.769879e-17 
                 SI.[umol/L]** AMODIS:PAR8d,Einsteins/m-2/d-1 
                 -2.286591e-17                   2.392930e-16 
                   Okubo-Weiss                  Lyapunov_exp. 
                  3.774379e-18                  -1.536978e-17 
                  grad_SST_adv                      retention 
                  2.928585e-17                   4.109207e-17 
     Mean.Depth.MLD.Sigma.[m]*       Mean.Depth.Max.Fluo.[m]* 
                  4.933666e-17                  -1.395423e-18 
        Mean.Depth.Max.N2.[m]*         Mean.Depth.Max.O2.[m]* 
                 -8.203744e-17                   1.307523e-17 
        Mean.Depth.Min.O2.[m]*     Mean.Depth.Nitracline.[m]* 
                  1.592245e-16                   1.875222e-17 
     miTAG.SILVA.Taxo.Richness    miTAG.SILVA.Phylo.Diversity 
                  2.786577e-16                   8.592770e-17 
              miTAG.SILVA.Chao                miTAG.SILVA.ace 
                  2.015995e-16                  -2.787529e-16 
           miTAG.SILVA.Shannon                     OG.Shannon 
                 -9.620652e-16                  -2.290157e-15 
                   OG.Richness                    OG.Evenness 
                 -4.218759e-16                   1.035089e-15 
  FC.-.heterotrophs.[cells/mL]     FC.-.autotrophs.[cells/mL] 
                 -4.392569e-17                   3.959053e-17 
      FC.-.bacteria.[cells/mL] FC.-.picoeukaryotes.[cells/mL] 
                  2.276719e-17                  -7.691059e-19 
   minimum.generation.time.[h] 
                 -3.314539e-17 </code></pre>
<pre class="r"><code>#head(df_meta_norm)
# colnames(df_meta_norm)

# select a subset of variables
envfit(pc, df_meta_norm, permu = 999, na.rm=T)</code></pre>
<pre><code>
***VECTORS

                                    PC1      PC2     r2 Pr(&gt;r)   
Mean_Lat*                       0.36148  0.93238 0.0169  0.738   
Mean_Long*                      0.41262  0.91090 0.3289  0.004 **
Mean_Depth.[m]*                 0.10180 -0.99481 0.0015  0.971   
Mean_Temperature.[deg.C]*       0.40533  0.91417 0.0261  0.621   
Mean_Salinity.[PSU]*            0.98347 -0.18108 0.0800  0.216   
Mean_Oxygen.[umol/kg]*          0.17124  0.98523 0.0066  0.906   
Mean_Nitrates[umol/L]*         -0.64669 -0.76275 0.0392  0.478   
NO2.[umol/L]**                  0.64700  0.76249 0.0269  0.644   
PO4.[umol/L]**                 -0.48354 -0.87532 0.0284  0.612   
NO2NO3.[umol/L]**              -0.62300 -0.78222 0.0467  0.423   
SI.[umol/L]**                   0.01987  0.99980 0.0192  0.693   
AMODIS:PAR8d,Einsteins/m-2/d-1 -0.34783  0.93756 0.0090  0.855   
Okubo-Weiss                     0.22448  0.97448 0.0930  0.147   
Lyapunov_exp.                  -0.24483  0.96956 0.0238  0.690   
grad_SST_adv                   -0.31121  0.95034 0.0134  0.779   
retention                      -0.11638 -0.99320 0.0153  0.730   
Mean.Depth.MLD.Sigma.[m]*      -0.04769 -0.99886 0.0518  0.368   
Mean.Depth.Max.Fluo.[m]*       -0.30722  0.95164 0.0022  0.961   
Mean.Depth.Max.N2.[m]*          0.39904 -0.91693 0.0300  0.575   
Mean.Depth.Max.O2.[m]*          0.43008  0.90279 0.0384  0.530   
Mean.Depth.Min.O2.[m]*          0.44600  0.89503 0.0242  0.606   
Mean.Depth.Nitracline.[m]*      0.74177  0.67065 0.0071  0.869   
miTAG.SILVA.Taxo.Richness      -0.15629 -0.98771 0.0696  0.263   
miTAG.SILVA.Phylo.Diversity    -0.13446 -0.99092 0.0408  0.482   
miTAG.SILVA.Chao               -0.13371 -0.99102 0.0285  0.604   
miTAG.SILVA.ace                -0.13907 -0.99028 0.0230  0.675   
miTAG.SILVA.Shannon            -0.26597 -0.96398 0.1396  0.045 * 
OG.Shannon                     -0.18849 -0.98208 0.0468  0.431   
OG.Richness                    -0.10987 -0.99395 0.0283  0.600   
OG.Evenness                    -0.05267  0.99861 0.0086  0.861   
FC.-.heterotrophs.[cells/mL]    0.42641  0.90453 0.0157  0.751   
FC.-.autotrophs.[cells/mL]      0.62152  0.78340 0.0629  0.321   
FC.-.bacteria.[cells/mL]        0.48167  0.87635 0.0258  0.646   
FC.-.picoeukaryotes.[cells/mL] -0.01083 -0.99994 0.0986  0.153   
minimum.generation.time.[h]     0.21868  0.97580 0.0077  0.888   
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
Permutation: free
Number of permutations: 999

99 observations deleted due to missingness</code></pre>
<pre class="r"><code>en &lt;- envfit(pc, df_meta_norm, permu = 999, na.rm=T)
en # here only longitude is significantly correlated!</code></pre>
<pre><code>
***VECTORS

                                    PC1      PC2     r2 Pr(&gt;r)   
Mean_Lat*                       0.36148  0.93238 0.0169  0.727   
Mean_Long*                      0.41262  0.91090 0.3289  0.002 **
Mean_Depth.[m]*                 0.10180 -0.99481 0.0015  0.972   
Mean_Temperature.[deg.C]*       0.40533  0.91417 0.0261  0.628   
Mean_Salinity.[PSU]*            0.98347 -0.18108 0.0800  0.207   
Mean_Oxygen.[umol/kg]*          0.17124  0.98523 0.0066  0.869   
Mean_Nitrates[umol/L]*         -0.64669 -0.76275 0.0392  0.476   
NO2.[umol/L]**                  0.64700  0.76249 0.0269  0.626   
PO4.[umol/L]**                 -0.48354 -0.87532 0.0284  0.588   
NO2NO3.[umol/L]**              -0.62300 -0.78222 0.0467  0.401   
SI.[umol/L]**                   0.01987  0.99980 0.0192  0.722   
AMODIS:PAR8d,Einsteins/m-2/d-1 -0.34783  0.93756 0.0090  0.816   
Okubo-Weiss                     0.22448  0.97448 0.0930  0.166   
Lyapunov_exp.                  -0.24483  0.96956 0.0238  0.687   
grad_SST_adv                   -0.31121  0.95034 0.0134  0.754   
retention                      -0.11638 -0.99320 0.0153  0.749   
Mean.Depth.MLD.Sigma.[m]*      -0.04769 -0.99886 0.0518  0.367   
Mean.Depth.Max.Fluo.[m]*       -0.30722  0.95164 0.0022  0.957   
Mean.Depth.Max.N2.[m]*          0.39904 -0.91693 0.0300  0.567   
Mean.Depth.Max.O2.[m]*          0.43008  0.90279 0.0384  0.513   
Mean.Depth.Min.O2.[m]*          0.44600  0.89503 0.0242  0.669   
Mean.Depth.Nitracline.[m]*      0.74177  0.67065 0.0071  0.882   
miTAG.SILVA.Taxo.Richness      -0.15629 -0.98771 0.0696  0.272   
miTAG.SILVA.Phylo.Diversity    -0.13446 -0.99092 0.0408  0.476   
miTAG.SILVA.Chao               -0.13371 -0.99102 0.0285  0.602   
miTAG.SILVA.ace                -0.13907 -0.99028 0.0230  0.666   
miTAG.SILVA.Shannon            -0.26597 -0.96398 0.1396  0.068 . 
OG.Shannon                     -0.18849 -0.98208 0.0468  0.404   
OG.Richness                    -0.10987 -0.99395 0.0283  0.592   
OG.Evenness                    -0.05267  0.99861 0.0086  0.864   
FC.-.heterotrophs.[cells/mL]    0.42641  0.90453 0.0157  0.744   
FC.-.autotrophs.[cells/mL]      0.62152  0.78340 0.0629  0.308   
FC.-.bacteria.[cells/mL]        0.48167  0.87635 0.0258  0.613   
FC.-.picoeukaryotes.[cells/mL] -0.01083 -0.99994 0.0986  0.141   
minimum.generation.time.[h]     0.21868  0.97580 0.0077  0.913   
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
Permutation: free
Number of permutations: 999

99 observations deleted due to missingness</code></pre>
<pre class="r"><code># plot(en)

# plot the vectors it into ordination plot (ggplot)
en_coord_cont = as.data.frame(scores(en, &quot;vectors&quot;)) * ordiArrowMul(en)

order(en$vectors$pvals) # order of p values</code></pre>
<pre><code> [1]  2 27 34 13  5 23 32 17 10 28  7 24 20 19  9 29 25 33  8  4 26 21 14 11  1
[26] 31 16 15 12 30  6 22 35 18  3</code></pre>
<pre class="r"><code>en_coord_cont[order(en$vectors$pvals),] # in the order of p values</code></pre>
<pre><code>                                        PC1         PC2
Mean_Long*                      0.339734341  0.75000000
miTAG.SILVA.Shannon            -0.142701204 -0.51720509
FC.-.picoeukaryotes.[cells/mL] -0.004880678 -0.45071028
Okubo-Weiss                     0.098304435  0.42674448
Mean_Salinity.[PSU]*            0.399353014 -0.07353063
miTAG.SILVA.Taxo.Richness      -0.059203047 -0.37415877
FC.-.autotrophs.[cells/mL]      0.223865147  0.28216926
Mean.Depth.MLD.Sigma.[m]*      -0.015578444 -0.32629698
NO2NO3.[umol/L]**              -0.193239647 -0.24262446
OG.Shannon                     -0.058524835 -0.30492900
Mean_Nitrates[umol/L]*         -0.183876264 -0.21687613
miTAG.SILVA.Phylo.Diversity    -0.038976730 -0.28724964
Mean.Depth.Max.O2.[m]*          0.121001667  0.25399943
Mean.Depth.Max.N2.[m]*          0.099193886 -0.22793171
PO4.[umol/L]**                 -0.117038564 -0.21186874
OG.Richness                    -0.026550535 -0.24020108
miTAG.SILVA.Chao               -0.032393732 -0.24008668
FC.-.bacteria.[cells/mL]        0.111144533  0.20221593
NO2.[umol/L]**                  0.152472694  0.17969005
Mean_Temperature.[deg.C]*       0.093938441  0.21186585
miTAG.SILVA.ace                -0.030266472 -0.21552317
Mean.Depth.Min.O2.[m]*          0.099538419  0.19975370
Lyapunov_exp.                  -0.054239362  0.21479201
SI.[umol/L]**                   0.003954288  0.19897168
Mean_Lat*                       0.067455430  0.17399063
FC.-.heterotrophs.[cells/mL]    0.076660238  0.16261669
retention                      -0.020642399 -0.17616552
grad_SST_adv                   -0.051700553  0.15787695
AMODIS:PAR8d,Einsteins/m-2/d-1 -0.047314248  0.12753364
OG.Evenness                    -0.007018969  0.13308604
Mean_Oxygen.[umol/kg]*          0.019926196  0.11464439
Mean.Depth.Nitracline.[m]*      0.089595294  0.08100496
minimum.generation.time.[h]     0.027503706  0.12272702
Mean.Depth.Max.Fluo.[m]*       -0.020748309  0.06426889
Mean_Depth.[m]*                 0.005589742 -0.05462599</code></pre>
<pre class="r"><code>en_coord &lt;- en_coord_cont[order(en$vectors$pvals),] %&gt;% head(10) # top 10 p values


# plot PCA
ggplot(df_merged,aes(x=PC1,y=PC2))+
  theme(plot.title = element_text(size = 20,hjust = 0.5, face=&quot;bold&quot;)) +
  geom_point(size = 5, alpha=0.9, aes(col= Depth))+
  scale_fill_manual(values=cols) +
  xlab(paste0(&#39;\n PC1&#39;, &quot; (explained variance: &quot;,round(eigs[1] / sum(eigs),3)*100,&quot;%)&quot;))+
  ylab(paste0(&quot;PC2&quot;,&quot; (explained variance: &quot;,round(eigs[2] / sum(eigs),3)*100,&quot;%) \n&quot;)) +
  ggtitle(&quot;Principle component analysis for all samples \n&quot;) +
  ## adjust positions
  guides(fill = guide_legend(ncol = 4,reverse = T))+
  theme(legend.position=&quot;bottom&quot;) +
  
  # vectors
  geom_segment(aes(x = 0, y = 0, xend = PC1, yend = PC2), 
       data = en_coord, size =1, alpha = 0.5, colour = &quot;grey30&quot;) +
  geom_text(data = en_coord, aes(x = PC1, y = PC2), colour = &quot;grey30&quot;,
       fontface = &quot;bold&quot;, label = row.names(en_coord)) +
  mytheme_2d</code></pre>
<p><img src="figure/220406_class_activity_import_data.Rmd/unnamed-chunk-5-1.png" width="1056" style="display: block; margin: auto;" /></p>
<p>##. 2.3. PCoA method Let’s use the Bray-Curtis dissimilarity * should someone do normalization before calculating pair-wise dissimilarity distance?<br />
<a href="https://ordnews.colostate.narkive.com/lMWF502c/1593-log-sqrt-and-other-transformation-with-bray-curtis-dissimilarity" class="uri">https://ordnews.colostate.narkive.com/lMWF502c/1593-log-sqrt-and-other-transformation-with-bray-curtis-dissimilarity</a></p>
<pre class="r"><code># calculate Bray-Curtis dissimilarity between samples
dist_BC &lt;- vegdist(t(df_x), &quot;bray&quot;) # transposed df_x
mat_BC &lt;- as.matrix(dist_BC)
dim(mat_BC)</code></pre>
<pre><code>[1] 139 139</code></pre>
<pre class="r"><code># PCA with prcomp
# transpose dataframe
pc &lt;- prcomp(mat_BC)
eigs &lt;- pc$sdev^2
# variance explained
eigs[1] / sum(eigs) # pc1</code></pre>
<pre><code>[1] 0.7717982</code></pre>
<pre class="r"><code>eigs[2] / sum(eigs) # pc2</code></pre>
<pre><code>[1] 0.118878</code></pre>
<pre class="r"><code># dataframe for plotting
df_out &lt;- as.data.frame(pc$x)
dim(df_out)</code></pre>
<pre><code>[1] 139 139</code></pre>
<pre class="r"><code>#head(df_out)

# add environmental variable
df_out &lt;- df_out %&gt;% select(PC1, PC2, PC3) # get only PC1, 2, 3
df_out &lt;- tibble::rownames_to_column(df_out, var = &quot;Sample.label&quot;)
dim(df_out)</code></pre>
<pre><code>[1] 139   4</code></pre>
<pre class="r"><code># get depth
# split sample label to get the depth SRF, DCM, MES
list_depth &lt;- str_split(df_out$Sample.label, &quot;_&quot;)
vec_depth &lt;- c()
for (i in 1:length(list_depth)){
  #print(i)
  # print(list_depth[[i]][3])
  vec_depth[i] &lt;- list_depth[[i]][3]
}
  
df_out$Depth &lt;- vec_depth
  
df_merged &lt;- df_out %&gt;% left_join(df_meta, by = c(&quot;Sample.label&quot;=&quot;Sample.label&quot;))
dim(df_merged)</code></pre>
<pre><code>[1] 139  42</code></pre>
<pre class="r"><code>colnames(df_merged)</code></pre>
<pre><code> [1] &quot;Sample.label&quot;                   &quot;PC1&quot;                           
 [3] &quot;PC2&quot;                            &quot;PC3&quot;                           
 [5] &quot;Depth&quot;                          &quot;PANGAEA.Sample.ID&quot;             
 [7] &quot;Mean_Date.[YY/MM/DD.hh:mm]*&quot;    &quot;Mean_Lat*&quot;                     
 [9] &quot;Mean_Long*&quot;                     &quot;Mean_Depth.[m]*&quot;               
[11] &quot;Mean_Temperature.[deg.C]*&quot;      &quot;Mean_Salinity.[PSU]*&quot;          
[13] &quot;Mean_Oxygen.[umol/kg]*&quot;         &quot;Mean_Nitrates[umol/L]*&quot;        
[15] &quot;NO2.[umol/L]**&quot;                 &quot;PO4.[umol/L]**&quot;                
[17] &quot;NO2NO3.[umol/L]**&quot;              &quot;SI.[umol/L]**&quot;                 
[19] &quot;AMODIS:PAR8d,Einsteins/m-2/d-1&quot; &quot;Okubo-Weiss&quot;                   
[21] &quot;Lyapunov_exp.&quot;                  &quot;grad_SST_adv&quot;                  
[23] &quot;retention&quot;                      &quot;Mean.Depth.MLD.Sigma.[m]*&quot;     
[25] &quot;Mean.Depth.Max.Fluo.[m]*&quot;       &quot;Mean.Depth.Max.N2.[m]*&quot;        
[27] &quot;Mean.Depth.Max.O2.[m]*&quot;         &quot;Mean.Depth.Min.O2.[m]*&quot;        
[29] &quot;Mean.Depth.Nitracline.[m]*&quot;     &quot;miTAG.SILVA.Taxo.Richness&quot;     
[31] &quot;miTAG.SILVA.Phylo.Diversity&quot;    &quot;miTAG.SILVA.Chao&quot;              
[33] &quot;miTAG.SILVA.ace&quot;                &quot;miTAG.SILVA.Shannon&quot;           
[35] &quot;OG.Shannon&quot;                     &quot;OG.Richness&quot;                   
[37] &quot;OG.Evenness&quot;                    &quot;FC.-.heterotrophs.[cells/mL]&quot;  
[39] &quot;FC.-.autotrophs.[cells/mL]&quot;     &quot;FC.-.bacteria.[cells/mL]&quot;      
[41] &quot;FC.-.picoeukaryotes.[cells/mL]&quot; &quot;minimum.generation.time.[h]&quot;   </code></pre>
<pre class="r"><code># plot PCoA
ggplot(df_merged,aes(x=PC1,y=PC2, label = Depth))+
  theme(plot.title = element_text(size = 20,hjust = 0.5, face=&quot;bold&quot;)) +
  geom_point(size = 5, alpha=0.9, aes(col= Depth))+
  scale_fill_manual(values=cols) +
  xlab(paste0(&#39;\n PCo1&#39;, &quot; (explained variance: &quot;,round(eigs[1] / sum(eigs),3)*100,&quot;%)&quot;))+
  ylab(paste0(&quot;PCo2&quot;,&quot; (explained variance: &quot;,round(eigs[2] / sum(eigs),3)*100,&quot;%) \n&quot;)) +
  ggtitle(&quot;Principle coordinate analysis (PCoA) for all samples \n&quot;) +
  ## adjust positions
  guides(fill = guide_legend(ncol = 4,reverse = T))+
  theme(legend.position=&quot;bottom&quot;) +
  mytheme_2d</code></pre>
<p><img src="figure/220406_class_activity_import_data.Rmd/unnamed-chunk-6-1.png" width="1056" style="display: block; margin: auto;" /></p>
</div>
<div id="regress-pco1-or-pco2-with-environmental-variables-or-traits" class="section level2">
<h2>2.3.1. Regress PCo1 or PCo2 with environmental variables or traits!!</h2>
<pre class="r"><code>## Let&#39;s use envfit or vectorfit function to fit environmental variable/trait to PC1 and PC2
# select a subset of variables
envfit(pc, df_meta_norm, permu = 999, na.rm=T)</code></pre>
<pre><code>
***VECTORS

                                    PC1      PC2     r2 Pr(&gt;r)  
Mean_Lat*                      -0.27431  0.96164 0.0142  0.771  
Mean_Long*                     -0.10426  0.99455 0.1711  0.032 *
Mean_Depth.[m]*                -0.27636 -0.96105 0.0188  0.704  
Mean_Temperature.[deg.C]*      -0.15339  0.98817 0.0195  0.686  
Mean_Salinity.[PSU]*           -0.61271  0.79031 0.0848  0.184  
Mean_Oxygen.[umol/kg]*          0.24388  0.96980 0.0279  0.602  
Mean_Nitrates[umol/L]*          0.11366 -0.99352 0.0340  0.518  
NO2.[umol/L]**                 -0.08086  0.99673 0.0362  0.510  
PO4.[umol/L]**                  0.12815 -0.99175 0.0131  0.766  
NO2NO3.[umol/L]**               0.16852 -0.98570 0.0314  0.547  
SI.[umol/L]**                   0.32750  0.94485 0.0065  0.887  
AMODIS:PAR8d,Einsteins/m-2/d-1  0.30155  0.95345 0.0231  0.671  
Okubo-Weiss                     0.03543  0.99937 0.0197  0.701  
Lyapunov_exp.                   0.49304  0.87001 0.0464  0.420  
grad_SST_adv                    0.04163 -0.99913 0.0294  0.567  
retention                      -0.39581 -0.91833 0.0189  0.718  
Mean.Depth.MLD.Sigma.[m]*      -0.22828 -0.97360 0.0447  0.431  
Mean.Depth.Max.Fluo.[m]*       -0.01371  0.99991 0.0019  0.969  
Mean.Depth.Max.N2.[m]*         -0.39775 -0.91749 0.0401  0.461  
Mean.Depth.Max.O2.[m]*         -0.96992  0.24341 0.0172  0.762  
Mean.Depth.Min.O2.[m]*         -0.43802 -0.89896 0.0103  0.831  
Mean.Depth.Nitracline.[m]*     -0.99331 -0.11550 0.0167  0.741  
miTAG.SILVA.Taxo.Richness      -0.16878 -0.98565 0.1083  0.116  
miTAG.SILVA.Phylo.Diversity    -0.19782 -0.98024 0.0748  0.217  
miTAG.SILVA.Chao               -0.20363 -0.97905 0.0675  0.267  
miTAG.SILVA.ace                -0.20585 -0.97858 0.0602  0.310  
miTAG.SILVA.Shannon            -0.07146 -0.99744 0.1294  0.079 .
OG.Shannon                     -0.10964 -0.99397 0.0366  0.495  
OG.Richness                    -0.25551 -0.96681 0.0462  0.374  
OG.Evenness                     0.42082  0.90714 0.0309  0.547  
FC.-.heterotrophs.[cells/mL]    0.05198  0.99865 0.0217  0.672  
FC.-.autotrophs.[cells/mL]     -0.07057  0.99751 0.0652  0.283  
FC.-.bacteria.[cells/mL]        0.01969  0.99981 0.0318  0.545  
FC.-.picoeukaryotes.[cells/mL] -0.70705 -0.70716 0.0101  0.797  
minimum.generation.time.[h]    -0.17236  0.98503 0.0036  0.954  
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
Permutation: free
Number of permutations: 999

99 observations deleted due to missingness</code></pre>
<pre class="r"><code>en &lt;- envfit(pc, df_meta_norm, permu = 999, na.rm=T)
en # here only longitude is significantly correlated!</code></pre>
<pre><code>
***VECTORS

                                    PC1      PC2     r2 Pr(&gt;r)  
Mean_Lat*                      -0.27431  0.96164 0.0142  0.774  
Mean_Long*                     -0.10426  0.99455 0.1711  0.038 *
Mean_Depth.[m]*                -0.27636 -0.96105 0.0188  0.704  
Mean_Temperature.[deg.C]*      -0.15339  0.98817 0.0195  0.707  
Mean_Salinity.[PSU]*           -0.61271  0.79031 0.0848  0.171  
Mean_Oxygen.[umol/kg]*          0.24388  0.96980 0.0279  0.573  
Mean_Nitrates[umol/L]*          0.11366 -0.99352 0.0340  0.516  
NO2.[umol/L]**                 -0.08086  0.99673 0.0362  0.512  
PO4.[umol/L]**                  0.12815 -0.99175 0.0131  0.795  
NO2NO3.[umol/L]**               0.16852 -0.98570 0.0314  0.554  
SI.[umol/L]**                   0.32750  0.94485 0.0065  0.901  
AMODIS:PAR8d,Einsteins/m-2/d-1  0.30155  0.95345 0.0231  0.639  
Okubo-Weiss                     0.03543  0.99937 0.0197  0.699  
Lyapunov_exp.                   0.49304  0.87001 0.0464  0.407  
grad_SST_adv                    0.04163 -0.99913 0.0294  0.549  
retention                      -0.39581 -0.91833 0.0189  0.714  
Mean.Depth.MLD.Sigma.[m]*      -0.22828 -0.97360 0.0447  0.417  
Mean.Depth.Max.Fluo.[m]*       -0.01371  0.99991 0.0019  0.963  
Mean.Depth.Max.N2.[m]*         -0.39775 -0.91749 0.0401  0.455  
Mean.Depth.Max.O2.[m]*         -0.96992  0.24341 0.0172  0.774  
Mean.Depth.Min.O2.[m]*         -0.43802 -0.89896 0.0103  0.832  
Mean.Depth.Nitracline.[m]*     -0.99331 -0.11550 0.0167  0.729  
miTAG.SILVA.Taxo.Richness      -0.16878 -0.98565 0.1083  0.103  
miTAG.SILVA.Phylo.Diversity    -0.19782 -0.98024 0.0748  0.229  
miTAG.SILVA.Chao               -0.20363 -0.97905 0.0675  0.255  
miTAG.SILVA.ace                -0.20585 -0.97858 0.0602  0.306  
miTAG.SILVA.Shannon            -0.07146 -0.99744 0.1294  0.073 .
OG.Shannon                     -0.10964 -0.99397 0.0366  0.491  
OG.Richness                    -0.25551 -0.96681 0.0462  0.420  
OG.Evenness                     0.42082  0.90714 0.0309  0.578  
FC.-.heterotrophs.[cells/mL]    0.05198  0.99865 0.0217  0.681  
FC.-.autotrophs.[cells/mL]     -0.07057  0.99751 0.0652  0.265  
FC.-.bacteria.[cells/mL]        0.01969  0.99981 0.0318  0.538  
FC.-.picoeukaryotes.[cells/mL] -0.70705 -0.70716 0.0101  0.828  
minimum.generation.time.[h]    -0.17236  0.98503 0.0036  0.944  
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
Permutation: free
Number of permutations: 999

99 observations deleted due to missingness</code></pre>
<pre class="r"><code># plot the vectors it into ordination plot (ggplot)
en_coord_cont = as.data.frame(scores(en, &quot;vectors&quot;)) * ordiArrowMul(en)

order(en$vectors$pvals) # order of p values</code></pre>
<pre><code> [1]  2 27 23  5 24 25 32 26 14 17 29 19 28  8  7 33 15 10  6 30 12 31 13  3  4
[26] 16 22  1 20  9 34 21 11 35 18</code></pre>
<pre class="r"><code>en_coord_cont[order(en$vectors$pvals),] # in the order of p values</code></pre>
<pre><code>                                        PC1         PC2
Mean_Long*                     -0.078623789  0.75000000
miTAG.SILVA.Shannon            -0.046866489 -0.65419529
miTAG.SILVA.Taxo.Richness      -0.101242946 -0.59125726
Mean_Salinity.[PSU]*           -0.325291497  0.41958439
miTAG.SILVA.Phylo.Diversity    -0.098604819 -0.48860185
miTAG.SILVA.Chao               -0.096438876 -0.46367301
FC.-.autotrophs.[cells/mL]     -0.032860282  0.46451232
miTAG.SILVA.ace                -0.092086317 -0.43776512
Lyapunov_exp.                   0.193640546  0.34169378
Mean.Depth.MLD.Sigma.[m]*      -0.088036005 -0.37546734
OG.Richness                    -0.100125565 -0.37886292
Mean.Depth.Max.N2.[m]*         -0.145209921 -0.33495806
OG.Shannon                     -0.038213726 -0.34644867
NO2.[umol/L]**                 -0.028062115  0.34591223
Mean_Nitrates[umol/L]*          0.038214791 -0.33403232
FC.-.bacteria.[cells/mL]        0.006398320  0.32485622
grad_SST_adv                    0.013013131 -0.31232845
NO2NO3.[umol/L]**               0.054465086 -0.31858361
Mean_Oxygen.[umol/kg]*          0.074223329  0.29514780
OG.Evenness                     0.134863038  0.29071737
AMODIS:PAR8d,Einsteins/m-2/d-1  0.083633149  0.26443476
FC.-.heterotrophs.[cells/mL]    0.013961237  0.26824635
Okubo-Weiss                     0.009072316  0.25590843
Mean_Depth.[m]*                -0.069062233 -0.24016344
Mean_Temperature.[deg.C]*      -0.039086838  0.25181143
retention                      -0.099301558 -0.23039538
Mean.Depth.Nitracline.[m]*     -0.233985895 -0.02720794
Mean_Lat*                      -0.059551961  0.20876740
Mean.Depth.Max.O2.[m]*         -0.231601518  0.05812124
PO4.[umol/L]**                  0.026778172 -0.20723598
FC.-.picoeukaryotes.[cells/mL] -0.129698340 -0.12971736
Mean.Depth.Min.O2.[m]*         -0.080852173 -0.16593508
SI.[umol/L]**                   0.048172570  0.13897827
minimum.generation.time.[h]    -0.018878878  0.10789368
Mean.Depth.Max.Fluo.[m]*       -0.001098432  0.08009149</code></pre>
<pre class="r"><code>en_coord &lt;- en_coord_cont[order(en$vectors$pvals),] %&gt;% head(10) # top 10 p values


# plot
ggplot(df_merged,aes(x=PC1,y=PC2))+
  theme(plot.title = element_text(size = 20,hjust = 0.5, face=&quot;bold&quot;)) +
  geom_point(size = 5, alpha=0.9, aes(col= Depth))+
  scale_fill_manual(values=cols) +
  xlab(paste0(&#39;\n PCo1&#39;, &quot; (explained variance: &quot;,round(eigs[1] / sum(eigs),3)*100,&quot;%)&quot;))+
  ylab(paste0(&quot;PCo2&quot;,&quot; (explained variance: &quot;,round(eigs[2] / sum(eigs),3)*100,&quot;%) \n&quot;)) +
  ggtitle(&quot;Principle component analysis for all samples \n&quot;) +
  ## adjust positions
  guides(fill = guide_legend(ncol = 4,reverse = T))+
  theme(legend.position=&quot;bottom&quot;) +
  
  # vectors
  geom_segment(aes(x = 0, y = 0, xend = PC1, yend = PC2), 
       data = en_coord, size =1, alpha = 0.5, colour = &quot;grey30&quot;) +
  geom_text(data = en_coord, aes(x = PC1, y = PC2), colour = &quot;grey30&quot;,
       fontface = &quot;bold&quot;, label = row.names(en_coord)) +
  mytheme_2d</code></pre>
<p><img src="figure/220406_class_activity_import_data.Rmd/unnamed-chunk-7-1.png" width="1056" style="display: block; margin: auto;" /></p>
</div>
<div id="nmds" class="section level2">
<h2>2.4. NMDS</h2>
<p>Again use Bray-Curtis distance</p>
<pre class="r"><code># Bray-Curtis distance
# dist_BC

# we`ll set a seed to make the results reproducible
set.seed(2)

NMDS &lt;- metaMDS(dist_BC, k = 2, trymax = 1000, trace = F)

stressplot(NMDS) # There is a good non-metric fit between observed dissimilarities (in our distance matrix) and the distances in ordination space.</code></pre>
<p><img src="figure/220406_class_activity_import_data.Rmd/unnamed-chunk-8-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df_nmds &lt;- as.data.frame(NMDS$points)
dim(df_nmds)</code></pre>
<pre><code>[1] 139   2</code></pre>
<pre class="r"><code>NMDS$stress</code></pre>
<pre><code>[1] 0.10903</code></pre>
<pre class="r"><code># get depth
# split sample lable to get the depth SRF, DCM, MES
list_depth &lt;- str_split(rownames(df_nmds), &quot;_&quot;)
vec_depth &lt;- c()
for (i in 1:length(list_depth)){
  #print(i)
  # print(list_depth[[i]][3])
  vec_depth[i] &lt;- list_depth[[i]][3]
}
  
df_nmds$Depth &lt;- vec_depth

# merge metadata
df_nmds &lt;- tibble::rownames_to_column(df_nmds, var=&quot;Sample.label&quot;)
df_merged &lt;- df_nmds %&gt;% left_join(df_meta, by = c(&quot;Sample.label&quot;=&quot;Sample.label&quot;))
dim(df_merged)</code></pre>
<pre><code>[1] 139  41</code></pre>
<pre class="r"><code>colnames(df_merged)</code></pre>
<pre><code> [1] &quot;Sample.label&quot;                   &quot;MDS1&quot;                          
 [3] &quot;MDS2&quot;                           &quot;Depth&quot;                         
 [5] &quot;PANGAEA.Sample.ID&quot;              &quot;Mean_Date.[YY/MM/DD.hh:mm]*&quot;   
 [7] &quot;Mean_Lat*&quot;                      &quot;Mean_Long*&quot;                    
 [9] &quot;Mean_Depth.[m]*&quot;                &quot;Mean_Temperature.[deg.C]*&quot;     
[11] &quot;Mean_Salinity.[PSU]*&quot;           &quot;Mean_Oxygen.[umol/kg]*&quot;        
[13] &quot;Mean_Nitrates[umol/L]*&quot;         &quot;NO2.[umol/L]**&quot;                
[15] &quot;PO4.[umol/L]**&quot;                 &quot;NO2NO3.[umol/L]**&quot;             
[17] &quot;SI.[umol/L]**&quot;                  &quot;AMODIS:PAR8d,Einsteins/m-2/d-1&quot;
[19] &quot;Okubo-Weiss&quot;                    &quot;Lyapunov_exp.&quot;                 
[21] &quot;grad_SST_adv&quot;                   &quot;retention&quot;                     
[23] &quot;Mean.Depth.MLD.Sigma.[m]*&quot;      &quot;Mean.Depth.Max.Fluo.[m]*&quot;      
[25] &quot;Mean.Depth.Max.N2.[m]*&quot;         &quot;Mean.Depth.Max.O2.[m]*&quot;        
[27] &quot;Mean.Depth.Min.O2.[m]*&quot;         &quot;Mean.Depth.Nitracline.[m]*&quot;    
[29] &quot;miTAG.SILVA.Taxo.Richness&quot;      &quot;miTAG.SILVA.Phylo.Diversity&quot;   
[31] &quot;miTAG.SILVA.Chao&quot;               &quot;miTAG.SILVA.ace&quot;               
[33] &quot;miTAG.SILVA.Shannon&quot;            &quot;OG.Shannon&quot;                    
[35] &quot;OG.Richness&quot;                    &quot;OG.Evenness&quot;                   
[37] &quot;FC.-.heterotrophs.[cells/mL]&quot;   &quot;FC.-.autotrophs.[cells/mL]&quot;    
[39] &quot;FC.-.bacteria.[cells/mL]&quot;       &quot;FC.-.picoeukaryotes.[cells/mL]&quot;
[41] &quot;minimum.generation.time.[h]&quot;   </code></pre>
<pre class="r"><code># plot NMDS
ggplot(df_merged, aes(x=MDS1,y=MDS2))+
  theme(plot.title = element_text(size = 20,hjust = 0.5, face=&quot;bold&quot;)) +
  geom_point(size = 5, alpha=0.9, aes(col= Depth))+
  scale_fill_manual(values=cols) +
  xlab(paste0(&#39;\n NMDS1&#39;))+
  ylab(paste0(&quot;NMDS2&quot;)) +
  ggtitle(&quot;Non-metric multidimensional scaling(NMDS) for all samples \n&quot;) +
  ## adjust positions
  guides(fill = guide_legend(ncol = 4,reverse = T))+
  theme(legend.position=&quot;bottom&quot;) +
  mytheme_2d</code></pre>
<p><img src="figure/220406_class_activity_import_data.Rmd/unnamed-chunk-8-2.png" width="1056" style="display: block; margin: auto;" /></p>
</div>
<div id="plot-with-envfit" class="section level1">
<h1>plot with envfit</h1>
<pre class="r"><code># select a subset of variables
envfit(NMDS, df_meta_norm, permu = 999, na.rm=T)</code></pre>
<pre><code>
***VECTORS

                                  NMDS1    NMDS2     r2 Pr(&gt;r)  
Mean_Lat*                      -0.27187  0.96233 0.0575  0.334  
Mean_Long*                     -0.43236 -0.90170 0.1848  0.028 *
Mean_Depth.[m]*                -0.04614  0.99893 0.0462  0.419  
Mean_Temperature.[deg.C]*      -0.26395 -0.96454 0.0487  0.438  
Mean_Salinity.[PSU]*           -0.35045 -0.93658 0.1568  0.051 .
Mean_Oxygen.[umol/kg]*          0.08053 -0.99675 0.0817  0.201  
Mean_Nitrates[umol/L]*          0.19861  0.98008 0.0854  0.223  
NO2.[umol/L]**                 -0.43260  0.90159 0.0436  0.443  
PO4.[umol/L]**                  0.11744  0.99308 0.0819  0.238  
NO2NO3.[umol/L]**               0.20326  0.97913 0.0875  0.205  
SI.[umol/L]**                   0.76522 -0.64377 0.0039  0.947  
AMODIS:PAR8d,Einsteins/m-2/d-1  0.07057  0.99751 0.0402  0.490  
Okubo-Weiss                    -0.29101 -0.95672 0.0154  0.753  
Lyapunov_exp.                   0.23574 -0.97182 0.0558  0.320  
grad_SST_adv                    0.17723  0.98417 0.0214  0.658  
retention                      -0.06318  0.99800 0.0169  0.722  
Mean.Depth.MLD.Sigma.[m]*      -0.21027 -0.97764 0.0064  0.889  
Mean.Depth.Max.Fluo.[m]*        0.09655 -0.99533 0.0116  0.829  
Mean.Depth.Max.N2.[m]*         -0.99439  0.10578 0.0221  0.667  
Mean.Depth.Max.O2.[m]*         -0.15609 -0.98774 0.0381  0.502  
Mean.Depth.Min.O2.[m]*         -0.06236  0.99805 0.0963  0.144  
Mean.Depth.Nitracline.[m]*     -0.40008 -0.91648 0.0066  0.906  
miTAG.SILVA.Taxo.Richness      -0.02130  0.99977 0.0811  0.191  
miTAG.SILVA.Phylo.Diversity    -0.05229  0.99863 0.0783  0.201  
miTAG.SILVA.Chao               -0.05424  0.99853 0.0633  0.274  
miTAG.SILVA.ace                -0.05376  0.99855 0.0555  0.327  
miTAG.SILVA.Shannon             0.27966  0.96010 0.0304  0.534  
OG.Shannon                     -0.00300  1.00000 0.0809  0.211  
OG.Richness                    -0.12826  0.99174 0.0554  0.297  
OG.Evenness                     0.41389 -0.91033 0.0201  0.685  
FC.-.heterotrophs.[cells/mL]   -0.99697  0.07783 0.0106  0.830  
FC.-.autotrophs.[cells/mL]     -0.20826 -0.97807 0.1337  0.080 .
FC.-.bacteria.[cells/mL]       -0.51719 -0.85587 0.0222  0.677  
FC.-.picoeukaryotes.[cells/mL] -0.19949 -0.97990 0.0236  0.681  
minimum.generation.time.[h]    -0.07443  0.99723 0.0750  0.244  
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
Permutation: free
Number of permutations: 999

99 observations deleted due to missingness</code></pre>
<pre class="r"><code>en_nmds &lt;- envfit(NMDS, df_meta_norm, permu = 999, na.rm=T)
en_nmds # here only longitude is significantly correlated!</code></pre>
<pre><code>
***VECTORS

                                  NMDS1    NMDS2     r2 Pr(&gt;r)  
Mean_Lat*                      -0.27187  0.96233 0.0575  0.307  
Mean_Long*                     -0.43236 -0.90170 0.1848  0.017 *
Mean_Depth.[m]*                -0.04614  0.99893 0.0462  0.418  
Mean_Temperature.[deg.C]*      -0.26395 -0.96454 0.0487  0.403  
Mean_Salinity.[PSU]*           -0.35045 -0.93658 0.1568  0.042 *
Mean_Oxygen.[umol/kg]*          0.08053 -0.99675 0.0817  0.208  
Mean_Nitrates[umol/L]*          0.19861  0.98008 0.0854  0.182  
NO2.[umol/L]**                 -0.43260  0.90159 0.0436  0.419  
PO4.[umol/L]**                  0.11744  0.99308 0.0819  0.191  
NO2NO3.[umol/L]**               0.20326  0.97913 0.0875  0.173  
SI.[umol/L]**                   0.76522 -0.64377 0.0039  0.923  
AMODIS:PAR8d,Einsteins/m-2/d-1  0.07057  0.99751 0.0402  0.454  
Okubo-Weiss                    -0.29101 -0.95672 0.0154  0.764  
Lyapunov_exp.                   0.23574 -0.97182 0.0558  0.327  
grad_SST_adv                    0.17723  0.98417 0.0214  0.671  
retention                      -0.06318  0.99800 0.0169  0.722  
Mean.Depth.MLD.Sigma.[m]*      -0.21027 -0.97764 0.0064  0.886  
Mean.Depth.Max.Fluo.[m]*        0.09655 -0.99533 0.0116  0.800  
Mean.Depth.Max.N2.[m]*         -0.99439  0.10578 0.0221  0.675  
Mean.Depth.Max.O2.[m]*         -0.15609 -0.98774 0.0381  0.513  
Mean.Depth.Min.O2.[m]*         -0.06236  0.99805 0.0963  0.139  
Mean.Depth.Nitracline.[m]*     -0.40008 -0.91648 0.0066  0.878  
miTAG.SILVA.Taxo.Richness      -0.02130  0.99977 0.0811  0.214  
miTAG.SILVA.Phylo.Diversity    -0.05229  0.99863 0.0783  0.228  
miTAG.SILVA.Chao               -0.05424  0.99853 0.0633  0.311  
miTAG.SILVA.ace                -0.05376  0.99855 0.0555  0.355  
miTAG.SILVA.Shannon             0.27966  0.96010 0.0304  0.570  
OG.Shannon                     -0.00300  1.00000 0.0809  0.218  
OG.Richness                    -0.12826  0.99174 0.0554  0.342  
OG.Evenness                     0.41389 -0.91033 0.0201  0.703  
FC.-.heterotrophs.[cells/mL]   -0.99697  0.07783 0.0106  0.802  
FC.-.autotrophs.[cells/mL]     -0.20826 -0.97807 0.1337  0.070 .
FC.-.bacteria.[cells/mL]       -0.51719 -0.85587 0.0222  0.640  
FC.-.picoeukaryotes.[cells/mL] -0.19949 -0.97990 0.0236  0.646  
minimum.generation.time.[h]    -0.07443  0.99723 0.0750  0.232  
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
Permutation: free
Number of permutations: 999

99 observations deleted due to missingness</code></pre>
<pre class="r"><code># plot the vectors it into ordination plot (ggplot)
en_coord_cont = as.data.frame(scores(en_nmds, &quot;vectors&quot;)) * ordiArrowMul(en_nmds)

order(en_nmds$vectors$pvals) # order of p values</code></pre>
<pre><code> [1]  2  5 32 21 10  7  9  6 23 28 24 35  1 25 14 29 26  4  3  8 12 20 27 33 34
[26] 15 19 30 16 13 18 31 22 17 11</code></pre>
<pre class="r"><code>en_coord_cont[order(en_nmds$vectors$pvals),] # in the order of p values</code></pre>
<pre><code>                                      NMDS1       NMDS2
Mean_Long*                     -0.449966599 -0.93842222
Mean_Salinity.[PSU]*           -0.335955325 -0.89785427
FC.-.autotrophs.[cells/mL]     -0.184391551 -0.86596587
Mean.Depth.Min.O2.[m]*         -0.046864159  0.75000000
NO2NO3.[umol/L]**               0.145546473  0.70112387
Mean_Nitrates[umol/L]*          0.140565160  0.69363129
PO4.[umol/L]**                  0.081361398  0.68797813
Mean_Oxygen.[umol/kg]*          0.055742786 -0.68999304
miTAG.SILVA.Taxo.Richness      -0.014681580  0.68914316
OG.Shannon                     -0.002064052  0.68870836
miTAG.SILVA.Phylo.Diversity    -0.035419527  0.67639267
minimum.generation.time.[h]    -0.049354619  0.66129791
Mean_Lat*                      -0.157833702  0.55868245
miTAG.SILVA.Chao               -0.033026762  0.60805515
Lyapunov_exp.                   0.134820035 -0.55578747
OG.Richness                    -0.073099564  0.56521631
miTAG.SILVA.ace                -0.030674640  0.56976064
Mean_Temperature.[deg.C]*      -0.141056349 -0.51545431
Mean_Depth.[m]*                -0.024000903  0.51957935
NO2.[umol/L]**                 -0.218603874  0.45560062
AMODIS:PAR8d,Einsteins/m-2/d-1  0.034261179  0.48426998
Mean.Depth.Max.O2.[m]*         -0.073755391 -0.46671308
miTAG.SILVA.Shannon             0.118080103  0.40538078
FC.-.bacteria.[cells/mL]       -0.186440756 -0.30853461
FC.-.picoeukaryotes.[cells/mL] -0.074199624 -0.36447050
grad_SST_adv                    0.062795261  0.34871157
Mean.Depth.Max.N2.[m]*         -0.358027473  0.03808759
OG.Evenness                     0.141945113 -0.31220147
retention                      -0.019890248  0.31416813
Okubo-Weiss                    -0.087515789 -0.28771228
Mean.Depth.Max.Fluo.[m]*        0.025186777 -0.25964009
FC.-.heterotrophs.[cells/mL]   -0.248290076  0.01938203
Mean.Depth.Nitracline.[m]*     -0.078400682 -0.17959668
Mean.Depth.MLD.Sigma.[m]*      -0.040699874 -0.18923289
SI.[umol/L]**                   0.116256091 -0.09780589</code></pre>
<pre class="r"><code>en_coord &lt;- en_coord_cont[order(en_nmds$vectors$pvals),] %&gt;% head(10) # top 10 p values

# plot NMDS
ggplot(df_merged, aes(x=MDS1,y=MDS2))+
  theme(plot.title = element_text(size = 20,hjust = 0.5, face=&quot;bold&quot;)) +
  geom_point(size = 5, alpha=0.9, aes(col= Depth))+
  scale_fill_manual(values=cols) +
  xlab(paste0(&#39;\n NMDS1&#39;))+
  ylab(paste0(&quot;NMDS2&quot;)) +
  ggtitle(&quot;Non-metric multidimensional scaling(NMDS) for all samples \n&quot;) +
  ## adjust positions
  guides(fill = guide_legend(ncol = 4,reverse = T))+
  theme(legend.position=&quot;bottom&quot;) +
  
  # vectors
  geom_segment(aes(x = 0, y = 0, xend = NMDS1, yend = NMDS2), 
       data = en_coord, size =1, alpha = 0.5, colour = &quot;grey30&quot;) +
  geom_text(data = en_coord, aes(x = NMDS1, y = NMDS2), colour = &quot;grey30&quot;,
       fontface = &quot;bold&quot;, label = row.names(en_coord)) +
  mytheme_2d</code></pre>
<p><img src="figure/220406_class_activity_import_data.Rmd/unnamed-chunk-9-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot(NMDS, type = &quot;t&quot;, display = &quot;sites&quot;)
plot(en, p.max = 0.3)</code></pre>
<p><img src="figure/220406_class_activity_import_data.Rmd/unnamed-chunk-9-2.png" width="1056" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 22000)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252 
[2] LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] gtools_3.8.2    reshape2_1.4.4  openxlsx_4.2.3  readxl_1.3.1   
 [5] magrittr_2.0.1  forcats_0.5.1   stringr_1.4.0   purrr_0.3.4    
 [9] readr_1.4.0     tidyr_1.1.3     tibble_3.0.4    tidyverse_1.3.1
[13] ape_5.5         vegan_2.5-7     lattice_0.20-41 permute_0.9-5  
[17] ggplot2_3.3.5   dplyr_1.0.5     workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] httr_1.4.2        sass_0.4.0        jsonlite_1.7.2    splines_4.0.3    
 [5] modelr_0.1.8      bslib_0.2.5.1     assertthat_0.2.1  highr_0.9        
 [9] cellranger_1.1.0  yaml_2.2.1        pillar_1.6.0      backports_1.2.1  
[13] glue_1.4.2        digest_0.6.27     promises_1.2.0.1  rvest_1.0.1      
[17] colorspace_2.0-0  htmltools_0.5.1.1 httpuv_1.6.0      Matrix_1.2-18    
[21] plyr_1.8.6        pkgconfig_2.0.3   broom_0.7.9       haven_2.4.1      
[25] scales_1.1.1      whisker_0.4       later_1.2.0       git2r_0.28.0     
[29] mgcv_1.8-33       farver_2.1.0      generics_0.1.0    ellipsis_0.3.2   
[33] withr_2.4.2       cli_3.0.1         crayon_1.4.1      evaluate_0.14    
[37] fs_1.5.0          fansi_0.4.2       nlme_3.1-149      MASS_7.3-53      
[41] xml2_1.3.2        tools_4.0.3       hms_1.1.0         lifecycle_1.0.0  
[45] munsell_0.5.0     reprex_2.0.0      cluster_2.1.0     zip_2.1.1        
[49] compiler_4.0.3    jquerylib_0.1.4   rlang_0.4.10      grid_4.0.3       
[53] rstudioapi_0.13   labeling_0.4.2    rmarkdown_2.9     gtable_0.3.0     
[57] DBI_1.1.1         R6_2.5.0          lubridate_1.7.10  knitr_1.37       
[61] utf8_1.1.4        rprojroot_2.0.2   stringi_1.5.3     parallel_4.0.3   
[65] Rcpp_1.0.5        vctrs_0.3.8       dbplyr_2.1.1      tidyselect_1.1.1 
[69] xfun_0.29        </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
